pvlib/iotools/era5.py

```diff
@@ -0,0 +1,196 @@
+import requests
+import pandas as pd
+from io import BytesIO, StringIO
+import zipfile
+import time
+
+
+VARIABLE_MAP = {
+    # short names
+    'd2m': 'temp_dew',
+    't2m': 'temp_air',
+    'sp': 'pressure',
+    'ssrd': 'ghi',
+    'tp': 'precipitation',
+
+    # long names
+    '2m_dewpoint_temperature': 'temp_dew',
+    '2m_temperature': 'temp_air',
+    'surface_pressure': 'pressure',
+    'surface_solar_radiation_downwards': 'ghi',
+    'total_precipitation': 'precipitation',
+}
+
+
+def same(x):
+    return x
+
+
+def k_to_c(temp_k):
+    return temp_k - 273.15
+
+
+def j_to_w(j):
+    return j / 3600
+
+
+def m_to_cm(m):
+    return m / 100
+
+
+UNITS = {
+    'u100': same,
+    'v100': same,
+    'u10': same,
+    'v10': same,
+    'd2m': k_to_c,
+    't2m': k_to_c,
+    'msl': same,
+    'sst': k_to_c,
+    'skt': k_to_c,
+    'sp': same,
+    'ssrd': j_to_w,
+    'strd': j_to_w,
+    'tp': m_to_cm,
+}
+
```

Comment:
I'd prefix all of this with `_` to discourage others from using it, even within pvlib.